packages/controllers/src/plugins/PluginController.ts

```diff
@@ -774,12 +765,12 @@ export class PluginController extends EventEmitter {
     property: keyof Plugin,
     value: unknown,
   ) {
-    const { plugins } = this.store.getState();
+    const { plugins } = this.state;
     const plugin = plugins[pluginName];
     const newPlugin = { ...plugin, [property]: value };
     const newPlugins = { ...plugins, [pluginName]: newPlugin };
-    this.updateState({
-      plugins: newPlugins,
+    this.update((state: any) => {
```

Comment:
Nit: probably worth commenting here about why we needed to use `any` - to avoid the incompatibility between Immer's `WritableState` type and recursive types like `Json`.

Reply:
Same for the other cases as well - anytime there's an update I suppose :thinking: